tests/graph_test.py

```diff
@@ -735,15 +735,67 @@ def test_setitem_with_mapped_operands_raises_on_conflict() -> None:
 
 def test_setitem_currently_does_not_allow_compatible_indices() -> None:
     g = nx.DiGraph()
-    g.add_edge('a', 'c')
-    g.add_edge('b', 'c')
+    g.add_edge('a', 'b')
+    g.add_edge('c', 'd')
 
     graph = cb.Graph(g)
-    mapped = graph.map({'a': [1, 2, 3], 'b': [11, 12, 13]}).reduce('c', name='d')
+    mapped1 = graph.map({'a': [1, 2, 3]})
+    mapped2 = graph['d'].map({'c': [11, 12, 13]}).reduce('d', name='e')
     # Note: This is a limitation of the current implementation. We could check if the
     # indices are identical and allow this. For simplicity we currently do not.
     with pytest.raises(ValueError, match="Conflicting new index names"):
-        mapped['x'] = mapped['d']
+        mapped1['x'] = mapped2
+
+
+@pytest.mark.parametrize(
+    'node_values',
+    [
+        {'a': [1, 2, 3], 'b': [11, 12, 13]},
+        {'a': np.array([1, 2, 3]), 'b': np.array([11, 12, 13])},
+        pd.DataFrame({'a': [1, 2, 3], 'b': [11, 12, 13]}),
+        {
+            'a': sc.array(dims=['x'], values=[1, 2, 3]),
+            'b': sc.array(dims=['x'], values=[11, 12, 13]),
+        },
```

Comment:
I think I'm a little confused because the name above is a `ScippDataArrayAdapter`, so I thought it would be a `sc.DataArray`, like the pandas `DataFrame` above.
But I guess you just need a mapping of key to ArrayLike? Does this mean that you never use the `.data` in the `DataArray`? 
I assume you need a structure that can be sliced/indexed. Could it be a `DataGroup` instead of a `DataArray` internally?

I realize this is besides the point of the current PR...

Reply:
`ScippDataArrayAdapter` also handles `scipp.Variable`, interpreting the latter as an data array without coords.

> I assume you need a structure that can be sliced/indexed. Could it be a DataGroup instead of a DataArray internally?

Using a `dict`currently, could in principle add support for `Dataset` and `DataGroup`. But not "instead of": The `DataArray` holds the values for a single node, the `dict` (or `DataGroup`, ...) maps to multiple nodes, just like the columns of `pandas.DataFrame` vs. its rows.